# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/sessionstore/SessionStoreListener.cpp
# Commit: 0b4595b2c153
# Full Hash: 0b4595b2c1531b83fc24fbf9b0f6ceacfc3506bd
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 09:35:42
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 4363e3a3d799 -r 0b4595b2c153 toolkit/components/sessionstore/SessionStoreListener.cpp
--- a/toolkit/components/sessionstore/SessionStoreListener.cpp	Wed Apr 22 01:37:52 2020 +0000
+++ b/toolkit/components/sessionstore/SessionStoreListener.cpp	Wed Apr 22 01:37:55 2020 +0000
@@ -487,7 +487,7 @@
 
   /* Collect data from all child frame */
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (auto& child : aBrowsingContext->GetChildren()) {
+  for (auto& child : aBrowsingContext->Children()) {
     aPositionDescendants[currentIdx] +=
         CollectPositions(child, aPositions, aPositionDescendants);
   }
@@ -576,7 +576,7 @@
 
   /* Collect data from all child frame */
   // This is not going to work for fission. Bug 1572084 for tracking it.
-  for (auto& child : aBrowsingContext->GetChildren()) {
+  for (auto& child : aBrowsingContext->Children()) {
     aInputs[currentIdx].descendants +=
         CollectInputs(child, aInputs, aIdVals, aXPathVals);
   }